<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TrainControllerUi</class>
 <widget class="QWidget" name="TrainControllerUi">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>908</width>
    <height>667</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>TrainControllerUi</string>
  </property>
  <widget class="QTabWidget" name="train_controller_tabs">
   <property name="geometry">
    <rect>
     <x>10</x>
     <y>10</y>
     <width>881</width>
     <height>641</height>
    </rect>
   </property>
   <property name="currentIndex">
    <number>0</number>
   </property>
   <widget class="QWidget" name="train_info_tab">
    <attribute name="title">
     <string>Train Info</string>
    </attribute>
    <widget class="QComboBox" name="train_index">
     <property name="geometry">
      <rect>
       <x>370</x>
       <y>20</y>
       <width>51</width>
       <height>31</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>18</pointsize>
      </font>
     </property>
     <property name="currentText">
      <string/>
     </property>
     <property name="currentIndex">
      <number>-1</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>10</y>
       <width>81</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>20</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Train:</string>
     </property>
    </widget>
    <widget class="QFrame" name="speed_frame">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>80</y>
       <width>281</width>
       <height>181</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <widget class="QLabel" name="actual_speed_label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Actual Speed</string>
      </property>
     </widget>
     <widget class="QLabel" name="commanded_speed_label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>111</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Commanded Speed</string>
      </property>
     </widget>
     <widget class="QLabel" name="setpoint_speed_label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>130</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Setpoint Speed</string>
      </property>
     </widget>
     <widget class="QLabel" name="speed_info_label">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>0</y>
        <width>111</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Speed Info</string>
      </property>
     </widget>
     <widget class="QLabel" name="actual_speed">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>50</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="commanded_speed">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>90</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="setpoint_speed">
      <property name="geometry">
       <rect>
        <x>130</x>
        <y>130</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="mph_label">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>50</y>
        <width>31</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Mph</string>
      </property>
     </widget>
     <widget class="QLabel" name="mph_label_2">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>90</y>
        <width>31</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Mph</string>
      </property>
     </widget>
     <widget class="QLabel" name="mph_label_3">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>130</y>
        <width>31</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Mph</string>
      </property>
     </widget>
    </widget>
    <widget class="QFrame" name="brake_frame">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>360</y>
       <width>281</width>
       <height>141</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>1</number>
     </property>
     <widget class="QLabel" name="brake_info_label">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>0</y>
        <width>111</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Brake Info</string>
      </property>
     </widget>
     <widget class="QLabel" name="service_brake_label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Service Brake</string>
      </property>
     </widget>
     <widget class="QLabel" name="emergency_brake_label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>90</y>
        <width>101</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Emergency Brake</string>
      </property>
     </widget>
     <widget class="QLabel" name="service_brake_status">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>50</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="emergency_brake_status">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>90</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="service_brake_button">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>50</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Toggle</string>
      </property>
     </widget>
     <widget class="QPushButton" name="emergency_brake_button">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>90</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Toggle</string>
      </property>
     </widget>
    </widget>
    <widget class="QFrame" name="cabin_info">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>80</y>
       <width>281</width>
       <height>291</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>1</number>
     </property>
     <widget class="QLabel" name="brake_info_label_2">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>0</y>
        <width>141</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Train Car Info</string>
      </property>
     </widget>
     <widget class="QPushButton" name="headlights_button">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>50</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Off</string>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="left_door_button">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>130</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Closed</string>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLabel" name="headlights_label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>50</y>
        <width>61</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Headlights</string>
      </property>
     </widget>
     <widget class="QLabel" name="left_door_label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>130</y>
        <width>52</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Left Door</string>
      </property>
     </widget>
     <widget class="QLabel" name="right_door_label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>170</y>
        <width>61</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Right Door</string>
      </property>
     </widget>
     <widget class="QPushButton" name="right_door_button">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>170</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Closed</string>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLabel" name="announcement_label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>210</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Announcements</string>
      </property>
     </widget>
     <widget class="QPushButton" name="announcement_button">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>210</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Announce</string>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLabel" name="cabin_lights_label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>90</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Cabin Lights</string>
      </property>
     </widget>
     <widget class="QPushButton" name="cabin_lights_button">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>90</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Off</string>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QPushButton" name="cabin_temp_button">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>250</y>
        <width>51</width>
        <height>23</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Set</string>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QLabel" name="cabin_temp_label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>250</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Cabin Temp</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="cabin_temp_value">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>250</y>
        <width>41</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="temp_units_label">
      <property name="geometry">
       <rect>
        <x>170</x>
        <y>250</y>
        <width>31</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>F</string>
      </property>
     </widget>
    </widget>
    <widget class="QFrame" name="manual_mode_frame">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>390</y>
       <width>281</width>
       <height>161</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>1</number>
     </property>
     <widget class="QLabel" name="manual_mode_label">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>0</y>
        <width>131</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Manual Mode</string>
      </property>
     </widget>
     <widget class="QLabel" name="password_label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Password</string>
      </property>
     </widget>
     <widget class="QLabel" name="set_setpoint_label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>100</y>
        <width>101</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Set Setpoint Speed</string>
      </property>
     </widget>
     <widget class="QPushButton" name="password_submit_button">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>50</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Submit</string>
      </property>
     </widget>
     <widget class="QPushButton" name="setpoint_speed_button">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>100</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Set</string>
      </property>
     </widget>
     <widget class="QLabel" name="manual_mode_status">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>0</y>
        <width>31</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Off</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="password">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>50</y>
        <width>113</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QSpinBox" name="setpoint_speed_edit">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>90</y>
        <width>61</width>
        <height>41</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QFrame" name="brake_frame_2">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>270</y>
       <width>281</width>
       <height>81</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>1</number>
     </property>
     <widget class="QLabel" name="authority_info_label">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>0</y>
        <width>131</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Authority Info</string>
      </property>
     </widget>
     <widget class="QLabel" name="authority_label">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>40</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Authority</string>
      </property>
     </widget>
     <widget class="QLabel" name="authority_status">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>40</y>
        <width>41</width>
        <height>21</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
    </widget>
    <widget class="QLabel" name="train_controller_label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>161</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Train Controller</string>
     </property>
    </widget>
    <zorder>train_index</zorder>
    <zorder>label_7</zorder>
    <zorder>speed_frame</zorder>
    <zorder>cabin_info</zorder>
    <zorder>brake_frame</zorder>
    <zorder>manual_mode_frame</zorder>
    <zorder>brake_frame_2</zorder>
    <zorder>train_controller_label</zorder>
   </widget>
   <widget class="QWidget" name="tab_2">
    <attribute name="title">
     <string>Engineer</string>
    </attribute>
    <widget class="QFrame" name="kp_ki_frame">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>120</y>
       <width>211</width>
       <height>151</height>
      </rect>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>1</number>
     </property>
     <widget class="QLabel" name="kp_ki_label">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>0</y>
        <width>81</width>
        <height>41</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>14</pointsize>
       </font>
      </property>
      <property name="text">
       <string>Kp &amp; Ki</string>
      </property>
     </widget>
     <widget class="QLabel" name="kp_label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>50</y>
        <width>31</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Kp</string>
      </property>
     </widget>
     <widget class="QLabel" name="ki_label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>100</y>
        <width>21</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Ki</string>
      </property>
     </widget>
     <widget class="QPushButton" name="kp_set_button">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>50</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Set</string>
      </property>
     </widget>
     <widget class="QPushButton" name="ki_set_button">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>100</y>
        <width>80</width>
        <height>23</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Set</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="kp_value">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>50</y>
        <width>21</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="ki_value">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>100</y>
        <width>21</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>0</string>
      </property>
     </widget>
    </widget>
    <widget class="QLabel" name="train_controller_label_2">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>161</width>
       <height>41</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>13</pointsize>
      </font>
     </property>
     <property name="text">
      <string>Train Controller</string>
     </property>
    </widget>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
